src/init/welcome.ts

```diff
@@ -5,25 +5,28 @@ import { getConfig } from '../utils/vscode-extensions';
 import { ISettingsManager, LangResourceKeys } from '../models';
 import { constants } from '../constants';
 import { activationCommand } from '../commands';
+import { extensionSettings } from '../settings/extensionSettings';
 
 const i18nManager = new LanguageResourceManager(vscode.env.language);
 
 export function manageWelcomeMessage(settingsManager: ISettingsManager): void {
   const themeName = getConfig().inspect(constants.vscode.iconThemeSetting).globalValue;
-  if (!settingsManager.getState().welcomeShown || themeName !== constants.extensionName) {
+
+  if (!settingsManager.getState().welcomeShown && themeName !== constants.extensionName) {
+    settingsManager.setWelcomeShownToTrue();
```

Comment:
Can you explain the case for this?

Reply:
Of course. Imagine this case:

- Some user install the extension.
- The sees the welcome message but he decides not to activate the extension yet.
- The closes the IDE.
- The user opens the IDE and receives again the welcome message.

Basically, this assures that this message is only shown once in the extension's lifetime.

Just to be clear, I decided to call for the translator friends because the message was not in discussion and the change was going to be implemented in a way or another. On the other hand, @jens1o, I'm under the impression that I didn't explain myself well enough as it was never intended to show two messages but only one with the version information in it. We all agree with one and only message, the thing that we're discussing is if we add the version to it, as we are already doing when any update happens. This is a one time message in extension's lifetime so no spoky things here! :wink: 

Finally, and following @JimiC reasoning about the utility that the version number has for the final user, then we could also apply this to the update message. It would have no meaning for the user either. I introduced this as a matter of consistency and because I feel that the version number can matter to some users.

Anyway, as I said yesterday, if you both feel that way I'm happy with our collective decision. I just wanted to present my case again for the last time.

Reply:
The thing is that the code flow already updates the status on first installation (https://github.com/vscode-icons/vscode-icons/blob/master/src/index.ts#L30). On first installation `settingsManager.isNewVersion()` is always `true`.

Reply:
Let me check this out because this was not happening to me. Maybe because while testing I didn't changed the version number.

Reply:
And that's exactly the case. On first installation `getState()` return the default state and `version` is `0.0.0`.

Reply:
You're absolutely right! It was because of the temp data file had the same version number. I'll roll it back.